### ISSUE TITLE
Reported error: accountData cannot be set on an Account

### ISSUE THREAD

**Issue by vladikoff** [mozilla/fxa-content-server#3466]
See report here: https://sentry.prod.mozaws.net/operations/fxa-content-product/group/223558/events/5923791/

```
/scripts/main.js  ? in s<.set< at line 32:19954
/scripts/main.js  ? in D at line 13:12921
/scripts/main.js  ? in s.partial/c at line 13:13444
/scripts/main.js  ? in b.Model at line 16:31774
/scripts/main.js  ? in J/d< at line 17:15787
/scripts/main.js  ? in l<.initAccount at line 32:22921
/scripts/main.js  ? in l<.getSignedInAccount at line 32:23293
/scripts/main.js  ? in l<.upgradeFromSession/< at line 32:25011
/scripts/main.js  ? in F at line 17:23170
/scripts/main.js  ? in E at line 17:23124
/scripts/main.js  ? in l at line 17:21172
```

![image](https://cloud.githubusercontent.com/assets/128755/12770228/bdd1e524-c9ec-11e5-86a3-df8109e24022.png)

**Comment by brettz9** [comment]
Getting this myself (when trying to log into Firefox Hello (Firefox Accounts) after a long time)... How to recover? After clicking the error, I get brought to Google...

**Comment by rfk** [comment]
There are several instances of this with slightly different tracebacks, e.g.:

  https://sentry.prod.mozaws.net/operations/fxa-content-product/group/223558/
  https://sentry.prod.mozaws.net/operations/fxa-content-product/group/237402/
  https://sentry.prod.mozaws.net/operations/fxa-content-product/group/223652/

But what they all have in common, is the error occurs when trying to load account data from localStorage.  Could it be that there's e.g. old data in localStorage in a different format to what we expect, and we're now rejecting it due to the improved validation of account model data?

**Comment by rfk** [comment]
Looks like the Account model did used to have an "accountData" attribute, long ago:

  https://github.com/mozilla/fxa-content-server/pull/1733/files#diff-f6f26d253984759fd1ff90518fbcde33R66

So what I suspect is happening here, is that these users are coming back to FxA for the first time in a long time, we're trying to load old localStorage data with unexpected account properties in it, and failing out.

It appears that this error is preventing people from signing in to sync, see e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1250451, so I'm going to bump the priority.

**Comment by rfk** [comment]
> Getting this myself (when trying to log into Firefox Hello (Firefox Accounts) after a long time)...How to recover?

@brettz9 you could try clearing the localStorage state for accounts by visiting https://accounts.firefox.com/clear in the affected browser.

**Comment by rfk** [comment]
The spike in the sentry graph in early feb suggests this was caused by train-55, which is indeed when we released the improved sandboxing of account model fields from https://github.com/mozilla/fxa-content-server/commit/d432199

**Comment by brettz9** [comment]
Thanks, that seems to have fixed it.

**Comment by shane-tomlinson** [comment]
From https://github.com/mozilla/fxa-bugzilla-mirror/issues/131

---------------------------


From https://bugzilla.mozilla.org/show_bug.cgi?id=1250451

User Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:44.0) Gecko/20100101 Firefox/44.0
Build ID: 20160209234642

Steps to reproduce:

1. Reset sync on computer 1
2. Sync fails on computer 2 (expected)
3a. Attempt to set the new password on computer 2.
3b. Use help page info to clear sync account from computer 2, attempt to log in to account with computer 2.
3c. Attempt to create new account with computer 2.



Actual results:

3a, 3b, 3c all produce HTTP 500 errors on multiple attempts spread over more than 24 hours.


Expected results:

3a/3b should have let me log in to my Sync account.
3c should have let me create a new Sync account.


------------------------------


The most relevant of the screenshots in the bug was this one: https://bugzilla.mozilla.org/attachment.cgi?id=8723476

**Comment by shane-tomlinson** [comment]
I can reproduce on stomlinson.dev.lcip.org. Looking in localStorage, I see old data with all kinds of extra fields:

```json
10:50:17.090 "{
  "7840b7deabbe40b59e4fc023cc097f0d": {
    "accountData": {
      "email": "set117@yahoo.com",
      "sessionToken": "9643f74d37871a572a053628109fa90f2c0e00274185bc26f391be13b0f1053b",
      "sessionTokenContext": null,
      "uid": "7840b7deabbe40b59e4fc023cc097f0d"
    },
    "assertion": {
      "_fxaClient": {
        "_client": {
          "request": {
            "baseUri": "https://stomlinson.dev.lcip.org/auth/v1",
            "timeout": 30000
          }
        },
        "_signUpResendCount": 0,
        "_passwordResetResendCount": 0,
        "_interTabChannel": {}
      },
      "_audience": "https://oauth-stomlinson.dev.lcip.org"
    },
    "oAuthClient": {},
    "profileClient": {
      "profileUrl": "https://stomlinson.dev.lcip.org/profile"
    },
    "fxaClient": {
      "_client": {
        "request": {
          "baseUri": "https://stomlinson.dev.lcip.org/auth/v1",
          "timeout": 30000
        }
      },
      "_signUpResendCount": 0,
      "_passwordResetResendCount": 0,
      "_interTabChannel": {}
    },
    "oAuthClientId": "ea3ca969f8c6bb0d",
    "uid": "7840b7deabbe40b59e4fc023cc097f0d",
    "email": "set117@yahoo.com",
    "sessionToken": "9643f74d37871a572a053628109fa90f2c0e00274185bc26f391be13b0f1053b",
    "sessionTokenContext": null,
    "lastLogin": 1416927764004
  }
}"

```

**Comment by shane-tomlinson** [comment]
Converting lastLogin to a readable date results in `2014-11-25T15:02:44.004Z`

**Comment by shane-tomlinson** [comment]
STR

1. Load up content server w/ sha 21e18a9
1. Sign in with an account.
1. Load up master
1. Load up FxA

**Comment by shane-tomlinson** [comment]
Digging a bit more, this problem manifests if a user signed/in before #1900 was merged. #1900 fixes #1975 (account.toJSON polutes localStorage with extra properties). If no actions were taken that caused the account to be updated in localStorage, then the data was kept around. If an action was taken that caused the account to be updated in localStorage, the extra data was removed. Post #3376, if the extra fields were read from localStorage and then an Account was created with that data, the `accountData cannot be set on an Account` error was thrown.
